cxx-sensors/src/main/java/org/sonar/cxx/sensors/compiler/gcc/CxxCompilerGccSensor.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Sonar C++ Plugin (Community)
+ * Copyright (C) 2010-2018 SonarOpenCommunity
+ * http://github.com/SonarOpenCommunity/sonar-cxx
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.cxx.sensors.compiler.gcc;
+
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.cxx.CxxLanguage;
+import org.sonar.cxx.CxxMetricsFactory;
+import org.sonar.cxx.sensors.compiler.CxxCompilerSensor;
+
+public class CxxCompilerGccSensor extends CxxCompilerSensor {
+
+  public static final String KEY = "GCC";
+  public static final String REPORT_PATH_KEY = "gcc.reportPath";
+  public static final String REPORT_REGEX_DEF = "gcc.regex";
+  public static final String REPORT_CHARSET_DEF = "gcc.charset";
+  public static final String DEFAULT_CHARSET_DEF = "UTF-8";
+  // search for single line with compiler warning message - order for groups: 1=file, 2=line, 3=message, 4=id
+  public static final String DEFAULT_REGEX_DEF = "^(.*):([0-9]+):[0-9]+:\\x20warning:\\x20(.*)\\x20\\[(.*)\\]$";
+  // ToDo: as long as java 7 API is not used the support of named groups for regular expression is not possible
+  // sample regex: "^.*[\\\\,/](?<filename>.*)\\((?<line>[0-9]+)\\)\\x20:\\x20warning\\x20(?<id>C\\d\\d\\d\\d):(?<message>.*)$";
+  // get value with e.g. scanner.match().group("filename");
+
+  public CxxCompilerGccSensor(CxxLanguage language) {
+    super(language, REPORT_PATH_KEY, CxxCompilerGccRuleRepository.getRepositoryKey(language),
+      new CxxCompilerGccParser());
+  }
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    descriptor
+      .name(getLanguage().getName() + " CxxCompilerGccSensor")
+      .onlyOnLanguage(getLanguage().getKey())
+      .createIssuesForRuleRepositories(getReportPathKey())
```

Comment:
**ERROR**: please use `getRuleRepositoryKey()` (embarrassingly I've introduced it by myself)